daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/package.scala

```diff
@@ -8,8 +8,35 @@ import com.daml.lf.data.Ref.PackageId
 import com.daml.lf.language.{Ast, LanguageVersion}
 import com.google.protobuf.CodedInputStream
 
+import scala.util.Using
+import scala.util.Using.Releasable
+import scala.util.control.NonFatal
+
 package object archive {
 
+  type Result[X] = Either[Error, X]
```

Comment:
```suggestion
  private[archive] type Result[X] = Either[Error, X]
```

Your choice, but aliases are expanded before testing for signature accessibility, so you may freely mark this as any privacy level even for public functions that use it, and getting `Result` out of the visible namespace might be a good idea.

Reply:
not a huge fan of the type synonyms for this type of stuff. You save very little typing and you have to keep chasing an indirection everywhere to make sense of the code.

Reply:
I drop the type signature in 88dfffda244f564c3e6202385a538285da2f5d49